tests/test_measures.py

```diff
@@ -1,177 +1,241 @@
 
 
 ##########
 # ZSCORE #
 ##########
 
+@pytest.mark.z_score
 def test_z_score(sample_dataframe):
+
     df = sample_dataframe
+    df_ams = am.calculate_measures(df, ['z_score'])
+    assert df_ams['z_score'][0] == 9.0
 
-    df['am'] = am.z_score(df)
-    assert df['am'][0] == 9.0
 
+@pytest.mark.z_score
 @pytest.mark.nan
 def test_z_score_nan(invalid_dataframe):
     df = invalid_dataframe
+    try:
+        am.calculate_measures(df, ['z_score'])
+    except ValueError:
+        pass
 
-    df['am'] = am.z_score(df)
-    assert np.isnan(df['am'][0])
 
+@pytest.mark.z_score
 @pytest.mark.zero
 def test_z_score_with_zeros(zero_dataframe):
     df = zero_dataframe
-
-    df['am'] = am.z_score(df)
-    assert np.isnan(df['am'][0])
+    df_ams = am.calculate_measures(df, ['z_score'])
+    assert isnan(df_ams['z_score'].iloc[0])
 
 
 #############
 # LOGLIKELI #
 #############
 
+@pytest.mark.log_likelihood
 def test_log_likelihood(sample_dataframe):
-    df = sample_dataframe
 
-    df['am'] = am.log_likelihood(df)
-    assert df['am'][0] == 65.01659467828966
+    df = sample_dataframe
+    df_ams = am.calculate_measures(df, ['log_likelihood'])
+    assert df_ams['log_likelihood'][0] == 65.01659467828966
 
 
+@pytest.mark.log_likelihood
 @pytest.mark.nan
 def test_log_likelihood_nan(invalid_dataframe):
     df = invalid_dataframe
-
-    df['am'] = am.log_likelihood(df)
-    assert np.isnan(df['am'][0])
+    try:
+        am.calculate_measures(df, ['log_likelihood'])
+    except ValueError:
+        pass
 
 
+@pytest.mark.log_likelihood
 @pytest.mark.zero
 def test_log_likelihood_with_zeros(zero_dataframe):
     df = zero_dataframe
-
-    df['am'] = am.log_likelihood(df)
-    assert df['am'][0] == 0.0
+    df_ams = am.calculate_measures(df, ['log_likelihood'])
+    assert isnan(df_ams['log_likelihood'].iloc[0])
 
 
 ######
 # hypergeometric-likelihood #
 ######
 
+@pytest.mark.hypergeometric_likelihood
 def test_hypergeometric_likelihood(sample_dataframe):
     df = sample_dataframe
 
-    df['am'] = am.hypergeometric_likelihood(df)
-    assert df['am'][0] == 8.938031057584301e-09
+    df = sample_dataframe
+    df_ams = am.calculate_measures(df, ['hypergeometric_likelihood'])
+    assert df_ams['hypergeometric_likelihood'][0] == 5.776904234533874e-14
 
 
+@pytest.mark.hypergeometric_likelihood
+@pytest.mark.nan
 def test_hypergeometric_likelihood_nan(invalid_dataframe):
     df = invalid_dataframe
-
-    df['am'] = am.hypergeometric_likelihood(df)
-    assert np.isnan(df['am'][0])
+    try:
+        am.calculate_measures(df, ['hypergeometric_likelihood'])
+    except ValueError:
+        pass
 
 
+@pytest.mark.hypergeometric_likelihood
+@pytest.mark.zero
 def test_hypergeometric_likelihood_with_zeros(zero_dataframe):
     df = zero_dataframe
-
-    df['am'] = am.hypergeometric_likelihood(df)
-    assert df['am'][0] == 1.0
-
-
-# Check if any warnings of errors are thrown. Might be an unstable test
-@pytest.mark.stability
```

Comment:
Why was this removed?

Reply:
random data may lead to ValueErrors (if e.g. O11 > N) – this test is thus not possible anymore. The alternative would be to return NAN values for wrong rows and throw a warning instead of an error. Do you prefer this?

Reply:
Let me have a closer look at it